firedrake/assemble.py

```diff
@@ -341,7 +341,11 @@ def get_map(x, bcs=None, decoration=None):
                 args.append(o.dat(op2.READ, get_map(o), flatten=True))
             for n in coeff_map:
                 c = coefficients[n]
-                args.append(c.dat(op2.READ, get_map(c), flatten=True))
+                if c.function_space():
+                    for c_ in c.split():
+                        args.append(c_.dat(op2.READ, get_map(c_), flatten=True))
+                else:
+                    args.append(c.dat(op2.READ, get_map(c), flatten=True))
```

Comment:
I guess the other way to have done this would be to augment `Constant` with `split`.